src/callbacks_step/timeseries.jl

```diff
@@ -0,0 +1,202 @@
+                                           timeseries_cache)
+
+  return DiscreteCallback(condition, timeseries_callback, save_positions=(false,false))
+end
+
+
+# Convenience constructor that unpacks the semidiscretization into mesh, equations, solver, cache
+function TimeseriesCallback(semi, point_coordinates; kwargs...)
+  mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+
+  return TimeseriesCallback(mesh, equations, solver, cache, point_coordinates; kwargs...)
+end
+
+
+# Convenience constructor that converts a vector of points into a Trixi-style coordinate array
+function TimeseriesCallback(mesh, equations, solver, cache, point_coordinates::AbstractVector;
+                            kwargs...)
+  # Coordinates are always stored in [ndims, n_points]
+  n_points = length(point_coordinates)
+  point_coordinates_ = Matrix{eltype(eltype(point_coordinates))}(undef, ndims(mesh), n_points)
+
+  for p in 1:n_points
+    for d in 1:ndims(mesh)
+      point_coordinates_[d, p] = point_coordinates[p][d]
+    end
+  end
+
+  return TimeseriesCallback(mesh, equations, solver, cache, point_coordinates_; kwargs...)
+end
+
+
+# This method is called as callback during the time integration.
+@inline function (timeseries_callback::TimeseriesCallback)(integrator)
+  @unpack iter = integrator
+  @unpack interval = timeseries_callback
+
+  # Create record if in correct interval (needs to be checked since the callback is also called
+  # after the final step for storing the data on disk, indepdendent of the current interval)
+  if integrator.iter % interval == 0
+    @timeit_debug timer() "time series" begin
+      # Store time and step
+      push!(timeseries_callback.time, integrator.t)
+      push!(timeseries_callback.step, iter)
+
+      # Unpack data
+      u_ode = integrator.u
+      semi = integrator.p
+      mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
+      u = wrap_array(u_ode, mesh, equations, solver, cache)
+
+      @unpack (point_data, solution_variables,
+              variable_names, timeseries_cache) = timeseries_callback
+
+      # Record state at points (solver/mesh-dependent implementation)
+      record_state_at_points!(point_data, u, solution_variables, length(variable_names), mesh,
+                              equations, solver, timeseries_cache)
+    end
+  end
+
+  # Store timeseries if this is the last time step
+  if isfinished(integrator)
+    @unpack (solution_variables, variable_names,
+             output_directory, filename,
+             point_coordinates, point_data,
+             time, step, timeseries_cache) = timeseries_callback
+    n_points = length(point_data)
+
+    h5open(joinpath(output_directory, filename), "w") do file
+      # Add context information as attributes
+      n_variables = length(variable_names)
+      attributes(file)["ndims"] = ndims(mesh)
+      attributes(file)["equations"] = get_name(equations)
+      attributes(file)["polydeg"] = polydeg(solver)
```

Comment:
That's very DG-specific...

Reply:
Thanks for catching this! Fixed in 87256c03.